react_add-todo-form Develop

DemetriodosAnjos opened the pull request:
[DEMO LINK] - https://DemetriodosAnjos.github.io/react_add-todo-form/

DemetriodosAnjos committed:
feat: implementar e corrigir componentes e configs para passar testes e padronizar lint

- Criar UserInfo, TodoInfo (com data-id) e TodoList
- Ajustar App.tsx e AddTodoForm.tsx conforme requisitos do page.spec.js
- Corrigir exports/index.ts e imports quebrados
- Configurar tsconfig.eslint.json e ajustar .eslintrc.js
- Revisar scripts npm para lint e format automáticos

mateacademy-ai-mentor commented on src/AddTodoForm.tsx:
```diff
@@ -0,0 +1,107 @@
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const handleSubmit = (e: React.FormEvent) => {
```
This violates checklist item #1 exactly: "1. [CODE STYLE] - Don't use one-letter variable naming. (It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search)". Change the handler parameter `e` to a descriptive name such as `event` in the form submit handler (`handleSubmit`).

mateacademy-ai-mentor commented on src/AddTodoForm.tsx:
```diff
@@ -0,0 +1,107 @@
+          type="text"
+          data-cy="titleInput"
+          value={title}
+          onChange={e => {
```
This violates checklist item #1 exactly: "1. [CODE STYLE] - Don't use one-letter variable naming. (It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search)". Rename the `e` parameter in the title input onChange to a descriptive name (for example `changeEvent`) and use `changeEvent.target.value`.

mateacademy-ai-mentor commented on src/AddTodoForm.tsx:
```diff
@@ -0,0 +1,107 @@
+          id="user-select"
+          data-cy="userSelect"
+          value={selectedUserId}
+          onChange={e => {
```
This violates checklist item #1 exactly: "1. [CODE STYLE] - Don't use one-letter variable naming. (It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search)". Rename the `e` parameter in the user select onChange to a descriptive name (for example `changeEvent`) and use `changeEvent.target.value`.

mateacademy-ai-mentor commented on src/App.tsx:
```diff
@@ -1,61 +1,48 @@
+
+  const addTodo = (newTodo: { title: string; userId: number }) => {
+    const maxId = Math.max(...todos.map(todo => todo.id), 0);
+    const user = users.find(u => u.id === newTodo.userId);
```
This violates checklist item #1: "1. [CODE STYLE] - Don't use one-letter variable naming." The `users.find` callback in `addTodo` uses the single-letter variable `u`. Rename it to a descriptive name such as `user` or `foundUser` to comply with the checklist and improve readability.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I’m requesting changes because there are explicit checklist violations that must be fixed before approval.

Critical issues to fix (checklist item #1: “Don’t use one-letter variable naming”):
- src/AddTodoForm.tsx
  - Line(s) with form submit handler: the parameter `e` in `handleSubmit` should be renamed to a descriptive name like `event` (e.g., `const handleSubmit = (event: React.FormEvent) => { event.preventDefault(); ... }`). This aligns with the checklist and improves readability.
  - Line(s) with title input onChange: the inline handler uses `e`. Rename the parameter to something descriptive like `changeEvent` and use `changeEvent.target.value`.
  - Line(s) with user select onChange: the inline handler uses `e`. Rename to `changeEvent` (or similar) and use `changeEvent.target.value`.
- src/App.tsx
  - In `addTodo`, the `users.find` callback uses `u` (around line 24). Rename it to `user` or `foundUser` for clarity and checklist compliance: `users.find(user => user.id === newTodo.userId)`.

Why this matters:
- This is an explicit checklist requirement, not a stylistic preference. Using descriptive names improves readability, searchability, and maintainability.

Minor suggestions (optional, not blocking):
- Consider extracting named handlers in AddTodoForm (e.g., `handleTitleChange`, `handleUserChange`) to reduce duplication and make code clearer.
- Optional sanitization of the title input (UA/EN letters, digits, spaces) if you wish to meet the optional requirement.

Everything else looks solid: functionality, validation flow, id generation, and rendering meet the requirements. Once you rename these variables, I’m confident this will be ready to approve. Keep up the good work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

DemetriodosAnjos committed:
refactor: substituir variáveis de uma letra por nomes descritivos e extrair manipuladores para melhorar legibilidade

mateacademy-ai-mentor commented on .eslintrc.cjs:
```diff
@@ -14,38 +14,43 @@ module.exports = {
     {
-      'files': ['**/*.spec.jsx'],
-      'rules': {
+      files: ['**/*.spec.jsx'],
```
The overrides.files pattern is set to ['**/*.spec.jsx'] but this project uses TypeScript/TSX specs (e.g., .spec.tsx). As written, this override likely never matches your test files. Update the pattern to include TypeScript test files (for example, add '**/*.spec.tsx' and/or '**/*.spec.ts'), or broaden the pattern to match both .jsx and .tsx.

mateacademy-ai-mentor commented on .eslintrc.cjs:
```diff
@@ -73,16 +82,22 @@
-      allowChildren: true,
-    }],
+    ],
+    'jsx-a11y/label-has-for': [
```
The rule `jsx-a11y/label-has-for` is deprecated in jsx-a11y and has been superseded by `jsx-a11y/label-has-associated-control`. Consider removing this deprecated rule or replacing it with supported rules. Relying on deprecated rules may lead to unexpected behavior with newer versions of the jsx-a11y plugin.

mateacademy-ai-mentor commented on cypress.config.ts:
```diff
@@ -22,5 +22,6 @@ module.exports = defineConfig({
       framework: 'react',
       bundler: 'vite',
     },
+    supportFile: 'cypress/support/component.ts',
```
The `supportFile` is set to 'cypress/support/component.ts', but the repository contains the component support file at 'src/cypress/support/component.ts'. This mismatch will prevent Cypress from loading the correct support file for component tests. Update this path to match the actual file location (e.g., 'src/cypress/support/component.ts') or move the support file to 'cypress/support/component.ts'.

mateacademy-ai-mentor commented on cypress.config.ts:
```diff
@@ -2,5 +1,6 @@
+export default defineConfig({
   e2e: {
     baseUrl: 'http://localhost:3000',
     specPattern: 'cypress/integration/**/*.spec.{js,ts,jsx,tsx}',
```
The e2e `specPattern` uses 'cypress/integration/**/...'. That is valid if your tests are under that directory, but note that in newer Cypress versions the default e2e directory is 'cypress/e2e'. If you intend to follow the modern layout, consider changing this pattern to 'cypress/e2e/**/*.spec.{js,ts,jsx,tsx}'.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

**Decision: REQUEST_CHANGES**

Your app’s core functionality and component structure look solid, but there are a couple of critical blockers that will fail CI or break the test setup. Please address the items below.

Critical blockers to fix:
1. ESLint failure (no-shadow) in `src/App.tsx`
   - Why: CI fails due to `@typescript-eslint/no-shadow` — variable `foundUser` is declared in overlapping scopes.
   - Where: `src/App.tsx`
     - In the `useMemo` mapping when doing `users.find(...)` you use a callback param named `foundUser`.
     - In `addTodo`, you also declare `const foundUser = ...`.
   - How to fix (pick one minimal change):
     - Rename the find callback parameter in `useMemo`:
       `user: users.find(user => user.id === todo.userId)`
     - Or rename the local variable in `addTodo`:
       `const selectedUser = users.find(user => user.id === newTodo.userId);`
   - After the change, run `npm run lint` to confirm it passes.

2. Cypress component support file path mismatch
   - Why: Cypress won’t load the component test support file; component tests may not start.
   - Where: `cypress.config.ts` — `component.supportFile` is set to `cypress/support/component.ts`.
     Actual file: `src/cypress/support/component.ts`.
   - How to fix: Update the path in `cypress.config.ts` to `supportFile: 'src/cypress/support/component.ts'` (or move the file to match the current config).

Important (non-blocking) improvements:
- `.eslintrc.cjs` rule key typo
  - Where: the rule key `implicit-arrow-linebreak:` has a trailing colon.
  - Why: ESLint ignores this intended rule.
  - How: Rename the key to `implicit-arrow-linebreak`.

- `.eslintrc.cjs` overrides pattern for test files
  - Where: `overrides.files` uses `['**/*.spec.jsx']`.
  - Why: Your tests are in TS/TSX, so the override likely never applies.
  - How: Include TS patterns, e.g. `['**/*.spec.{ts,tsx,jsx,js}']` or add `**/*.spec.tsx` and `**/*.spec.ts`.

- Deprecated a11y rule
  - Where: `.eslintrc.cjs` — `jsx-a11y/label-has-for`.
  - Why: Deprecated; use modern equivalent.
  - How: Replace with `jsx-a11y/label-has-associated-control` or remove the deprecated rule.

What’s good:
- The React/TS implementation, form validation flow, data preparation in App, and rendering via TodoList all meet the task requirements. The UI logic, IDs, and accessibility considerations are well done.

Please fix the two blockers above so CI can pass. Once those are resolved, everything else looks on track. Keep up the good work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

DemetriodosAnjos committed:
refactor: renomear variáveis de uma letra para nomes descritivos, extrair manipuladores e adicionar validação extra no título

DemetriodosAnjos committed:
chore(eslint): incluir padrões de teste TypeScript e atualizar regra obsoleta jsx-a11y

- Ajusta o padrão de arquivos em overrides para abranger  e , garantindo que testes em TypeScript sejam corretamente identificados.
- Substitui a regra obsoleta  pela nova , conforme recomendação do plugin.
- Remove dependência de regra descontinuada para evitar comportamentos inesperados em versões futuras.

pedro-ruas reviewed (changes requested):
Demetrio, before approving this, I need to be sure you know what you are doing, so I have to ask questions:
- Why are there cypress changes in your commit?
- Why did you change tsconfig files?
- Why did you change all styling if not asked in the checklist?
- Why are there so many comments in your files? And why are they in Portuguese?

pedro-ruas reviewed (changes requested):
You did not answer any of the questions I asked on my last review.

DemetriodosAnjos commented:
> * Why are there cypress changes in your commit?
> * Why did you change tsconfig files?
> * Why did you change all styling if not asked in the checklist?
> * Why are there so many comments in your files? And why are they in Portuguese?

O projeto apresentou algumas inconsistências nos ítens citados acima...por isso julguei ser conveniente realizar as alterações. Realizei os testes e funcionaram.  Sobre os comentários...tenho hábito de comentar os códigos para fixar a compreensão sobre o assunto. @pedro-ruas

pedro-ruas commented:
O que queres dizer com "inconsistências"? Isso não responde todas as perguntas...
> > * Why are there cypress changes in your commit?
> > * Why did you change tsconfig files?
> > * Why did you change all styling if not asked in the checklist?
> > * Why are there so many comments in your files? And why are they in Portuguese?
> 
> O projeto apresentou algumas inconsistências nos ítens citados acima...por isso julguei ser conveniente realizar as alterações. Realizei os testes e funcionaram.  Sobre os comentários...tenho hábito de comentar os códigos para fixar a compreensão sobre o assunto. @pedro-ruas 
>
